modules/indexer/code/elasticsearch/elasticsearch.go

```diff
@@ -180,11 +180,18 @@ func (b *Indexer) Index(ctx context.Context, repo *repo_model.Repository, sha st
 	}
 
 	if len(reqs) > 0 {
-		_, err := b.inner.Client.Bulk().
-			Index(b.inner.VersionedIndexName()).
-			Add(reqs...).
-			Do(ctx)
-		return err
+		esBatchSize := 50
+		bulkService := b.inner.Client.Bulk().
+			Index(b.inner.VersionedIndexName())
```

Comment:
Great! The last question. Whether the `Index` will be reset after `Do` invoked? I'm not familiar with ES. But to ensure it's right, maybe moving this line in the loop is better.

Reply:
You're right! moving the `Index` and `Bulk` in loop is better option to ensure the consistency (across shards *if any) in case we've more than 1 nodes in ElasticSearch cluster. 've filed PR